### PR TITLE
[fix]: [PL-22647]: Fixed error handling for file with same name

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
Fixed error handling for file with same name

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30893)

### CODE REVIEW

File: 900-git-sync-sdk/src/test/java/io/harness/gitsync/scm/SCMGitSyncHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    when(gitSyncSdkService.isDefaultBranch(entityDetail.getEntityRef().getAccountIdentifier(),
+             entityDetail.getEntityRef().getOrgIdentifier(), entityDetail.getEntityRef().getProjectIdentifier()))
+        .thenReturn(true);
+    ScmPushResponse scmPushResponse1 = scmGitSyncHelper.pushToGit(gitEntityInfo1, yaml, ChangeType.ADD, entityDetail);
+    assertThat(scmPushResponse1.isPushToDefaultBranch()).isEqualTo(true);
+
+    Principal principal = new UserPrincipal(name, "DUMMY_USER_EMAIL", name, accountId);
+    SourcePrincipalContextBuilder.setSourcePrincipal(principal);
+    when(entityDetailRestToProtoMapper.createEntityDetailDTO(entityDetail)).thenReturn(entityDetailProtoDTO);
+    MDC.setContextMap(contextMap);
```
Comment (abhinav-harness) [inline 900-git-sync-sdk/src/test/java/io/harness/gitsync/scm/SCMGitSyncHelperTest.java:113]: unset this too

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
one change

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30893)* status: 3 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java, line 65 at r1](https://reviewable.io/reviews/harness/harness-core/30893#-MwWfyxK-joJht7f6ZyD:-MwWfyxL2a0FrGjO9gvX:b-grkke1) ([raw file](https://github.com/harness/harness-core/blob/15f9160d578d1ec3e7f0dcf986f7f8adfdc315bc/900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java#L65)):*
> ```Java
>     try {
>       checkForError(pushFileResponse);
>     } catch (WingsException e) {
> ```

Whyy this change ?

**Comment by rathodmeetsatish** [comment]
trigger messagecheck

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30893)* status: 3 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[900-git-sync-sdk/src/test/java/io/harness/gitsync/scm/SCMGitSyncHelperTest.java, line 113 at r1](https://reviewable.io/reviews/harness/harness-core/30893#-MwRTYlqtvs_oRuqCPM7-r1-113:-Mw_QX7c-HU1-lFNOa3Z:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/15f9160d578d1ec3e7f0dcf986f7f8adfdc315bc/900-git-sync-sdk/src/test/java/io/harness/gitsync/scm/SCMGitSyncHelperTest.java#L113)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

unset this too 
</blockquote></details>

Done.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30893)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by rijajoo — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30893#-:-Mw_rqy--rAuQp5tk8pW:bnfp4nl)* status: 1 change requests. 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30893#-:-Mw_sxvUDuOWjiNpx8Zs:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**